File: LifecycleApp/app/src/main/java/com/example/lifecycleapp/MainActivity.java
package com.example.lifecycleapp;


import android.app.Activity;
import android.os.Bundle;
import android.util.Log;

import androidx.annotation.Nullable;

public class MainActivity extends Activity {

    public static final String ACTIVITY_NAME = "MainActivity";

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Log.d(ACTIVITY_NAME, "onCreate");
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.d(ACTIVITY_NAME, "onStart");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.d(ACTIVITY_NAME, "onResume");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.d(ACTIVITY_NAME, "onPause");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.d(ACTIVITY_NAME, "onStop");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.d(ACTIVITY_NAME, "onDestroy");
    }
}

File: LifecycleApp/app/src/main/java/com/example/lifecycleapp/MainActivity.kt
package com.example.lifecycleapp

import android.os.Bundle
import android.os.PersistableBundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    val ACTIVITY_NAME = "MainActivity"
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.layout_main)
    }

    override fun onStart() {
        super.onStart()
        Log.d(ACTIVITY_NAME, "onStart")
    }

    override fun onResume() {
        super.onResume()
        Log.d(ACTIVITY_NAME, "onResume")
    }

    override fun onPause() {
        super.onPause()
        Log.d(ACTIVITY_NAME, "onPause")
    }

    override fun onStop() {
        super.onStop()
        Log.d(ACTIVITY_NAME, "onStop")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d(ACTIVITY_NAME, "onDestroy")
    }



}
